### PR TITLE
[Doc] Add SPV_INTEL_non_constant_addrspace_printf spec

### PR DESCRIPTION
**Pull request by KornevNikita**

### CODE REVIEW

**Comment by KornevNikita** [comment]
@AlexeySotkin @bashbaug @mbelicki I've marked this PR as a draft, but it is ready for review. The only thing - there is no opcode for the new capability yet, so we shouldn't merge it even if there are no concerns.

**Comment by KornevNikita** [comment]
@AlexeySotkin @bashbaug @mbelicki please take a look

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:46]: Please set the date explicitly. Somehow this attribute always show the current date.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:160]: Please use explicit date as well.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:136]: The spec has a pre-defined [format ](https://www.khronos.org/registry/SPIR-V/specs/unified1/OpenCL.ExtendedInstructionSet.100.html#_binary_form)for enumerating address spaces for pointers. There is no "or" or "address space".

```suggestion
_format_ must be a _pointer(constant, generic, private, local, global)_ to _i8_.
```

**Comment by mbelicki** [comment]
If I understand correctly the only change to the original extension is:

    _format_ must be a _pointer(constant, generic, private, local, global)_ to _i8_.

i.e. allowing the format pointer to be in other than constant address space. This is aligned with what was previously discussed and supported by current implementation, but in this case I think that we also need to specify that the format string must be a string literal, known at compile time. Otherwise this extension allows any i8 pointer to be a valid printf format argument.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:131]: I think phrase "The format string must be a string literal, known at compile time." is not quite accurate. I suggest replacing it with slightly changed words from [OpenCL C spec](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_C.html#printf-format-string).

> The format string must be resolvable at compile time, i.e. cannot be dynamically created by the executing program itself.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:23]: Should we add Nikita as a contributor?

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:26]:
```suggestion
Copyright (c) 2022 Intel Corporation.  All rights reserved.
```

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (KornevNikita) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc]: Thanks, resolved in e868c5f

**Review by AlexeySotkin — APPROVED** [review]

**Review by AlexeySotkin — APPROVED** [review]

**Comment by AlexeySotkin** [comment]
@mbelicki , @bashbaug ping.

**Review by mbelicki — DISMISSED** [review]
Looks good to me!

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (bader) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:48]: I would prefer using numbers for revision as we do for SYCL extensions.
```suggestion
| Revision           | 1
```

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (bader) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:57]: The latest available revision is 7. Should we update to the latest reversion?

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (bader) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc:54]: The latest available revision is 6. Should we update to the latest reversion?

**Review by bader — APPROVED** [review]
LGTM, just a few minor comments which can be ignored.
Please, let me know if you are willing to address them or we need to get approval from Ben to merge it.

**Review by bashbaug — DISMISSED** [review]
LGTM, I've verified that the token assignments are correct.

As I've mentioned offline I'm not thrilled with the extension and capability name, but I can't think of an name that's obviously better and I can live with the current names.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc
Comment (KornevNikita) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_non_constant_addrspace_printf.asciidoc]: Thanks, applied in 906a4cd

**Pull request merged**
